src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs

```diff
@@ -77,6 +79,8 @@ void ExecuteTransportOperation(TransportTransaction transaction, UnicastTranspor
 
                         var label = GetLabel(message);
 
+                        await Task.Yield();
```

Comment:
I was more considering the best/optimal usage. For example, you can make it the first statements in a method because you know the code will hit a blocking call as you suggest or you put it just before the blocking call. Not sure when you would want to choose one over the other.

Also, I've read that `Task.Yield` is using context synchronization. You cannot do `.ConfigureAwait(false)` on it and an alternative would be to use `await Task.Run(..).ConfigureAwait(false)` but after testing that it didn't make a difference while the code also becomes more complex due to the lambda.